### ISSUE TITLE
After enabling cache: "Unable to connect to the elevated service."

### ISSUE THREAD

**Issue by Christoph-Wagner** [gerardog/gsudo#40]
I installed gsudo 0.7 with chocolatey. Testing it, everything works fine. But after I set Cache to Auto, I get `Error: Unable to connect to the elevated service.` on every invocation. Restarting the computer did not help. Powershell or cmd makes no difference.

After clearing the registry key, I tried `gsudo cache on`, this does not result in the error, but it also doesn’t work, any request afterwards results in a new UAC prompt. Auto or Explicit doesn’t make a difference.

Windows 10 V2004 Build 19041.264 (VM, German)
Windows 10 V1909 Build 18363.836 (German, English language pack)

**Comment by gerardog** [comment]
Hi, could you please run 'gsudo -k' to stop all cache sessions,  then retry the whole operation with 'gsudo --debug'. A second Windows will open. Please paste both windows contents here. That will help me find out whats going on. 
Thanks.

**Comment by Christoph-Wagner** [comment]
Commands:

```
PS C:\Users\DM2> gsudo
Error: Unable to connect to the elevated service.
PS C:\Users\DM2> gsudo -k
Info: No active credentials found to invalidate.
PS C:\Users\DM2> gsudo --debug
Debug: IsWindowsApp(""C:\Program Files\PowerShell\7\pwsh.exe"") = False ("C:\Program Files\PowerShell\7\pwsh.exe")
Debug: Command to run: "C:\Program Files\PowerShell\7\pwsh.exe" -NoLogo
Debug: Using Console mode TokenSwitch
Debug: Caller PID: 18736
Debug: Caller SID: S-1-5-21-2780479403-2581287647-19755949-1002
Debug: Elevating process: C:\ProgramData\chocolatey\lib\gsudo\bin\gsudo.exe --debug gsudoservice 18736 S-1-5-21-2780479403-2581287647-19755949-1002 All 00:05:00
Debug: Elevated instance started.
Error: Unable to connect to the elevated service.
```

Debug:

```
Info: Service started
Debug: Service will shutdown if idle for 00:05:00
Error: System.Security.Principal.IdentityNotMappedException: Some or all identity references could not be translated.
   at System.Security.Principal.NTAccount.Translate(IdentityReferenceCollection sourceAccounts, Type targetType, Boolean forceSuccess)
   at System.Security.Principal.NTAccount.Translate(Type targetType)
   at System.Security.AccessControl.CommonObjectSecurity.ModifyAccess(AccessControlModification modification, AccessRule rule, Boolean& modified)
   at System.Security.AccessControl.CommonObjectSecurity.AddAccessRule(AccessRule rule)
   at gsudo.Rpc.NamedPipeServer.<Listen>d__16.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at gsudo.Commands.ServiceCommand.<Execute>d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at gsudo.Program.<Start>d__1.MoveNext()
Press any key to exit.
```

**Comment by gerardog** [comment]
I guess it's failing [here](https://github.com/gerardog/gsudo/blob/f7a65e3897957e2aaded9669894cc07730f741c4/src/gsudo/Rpc/NamedPipeServer.cs#L65) to map 'NT Authority\Network' to a valid SID due to the OS language. I should use it's SID instead there.

**Comment by Christoph-Wagner** [comment]
Yes, you are right. I just checked and changing that string to "NT-AUTORITÄT\NETZWERKDIENST" makes it work.

**Comment by gerardog** [comment]
Great! This should work on all languages: 
'new SecurityIdentifier("S-1-5-2")', but I don't have my machine to test it right now.

**Comment by Christoph-Wagner** [comment]
It does :)

**Comment by gerardog** [comment]
FYI: The fix was released in [v0.7.1](https://github.com/gerardog/gsudo/releases/tag/v0.7.1).
Thanks for reporting the problem.

**Comment by gerardog** [comment]
Cgf zz- -.a
adtgfczvn-.
- 
.8fddghgjhfhgyjti5225i714uytg- . -
